ddp-workspace/projects/ddp-dsm-ui/src/app/FON/pages/participantsList/participantsList.component.ts

```diff
@@ -15,18 +17,23 @@ export class ParticipantsListComponent implements OnInit {
   patients$: Observable<patientListModel[]>;
   totalCount$: Observable<number>;
   loading$: Observable<boolean>;
+  error$: Observable<string>;
+
   pageIndex: number;
   pageSize: number;
+  readonly PARENT = MainConstants.participantsListParent;
 
   private readonly LSParams: string = 'pListQueryParams';
 
-  constructor(private agent: AgentService, private router: Router, private activatedRoute: ActivatedRoute) {
+  constructor(private router: Router, private activatedRoute: ActivatedRoute, private storeService: StoreService) {
   }
 
   ngOnInit(): void {
-    this.patients$ = this.agent.getPatients();
-    this.totalCount$ = this.agent.getPatientsTotalCount();
-    this.loading$ = this.agent.isLoading();
+    this.totalCount$ = this.storeService.getParticipantsTotalCount;
```

Comment:
This is how we assign observable selectors to local observable properties in order to observe changes in store, therefore this component always stays informed about the latest changes.